sig-scalability/slis/apimachinery_slis.md

```diff
@@ -0,0 +1,199 @@
+# API-machinery SLIs and SLOs
+
+The document was converted from [Google Doc]. Please refer to the original for
+extended commentary and discussion.
+
+***TODO(wojtek-t,deads2k,lavalamp): Get to consensus if we want to define latency
+as "to last byte" or as "to first byte"***
+
+## Background
+
+Scalability is an important aspect of the Kubernetes. However, Kubernetes is
+such a large system that we need to manage users expectations in this area.
+To achieve it, we are in process of redefining what does it mean that
+Kubernetes supports X-node clusters - this doc describes the high-level
+proposal. In this doc we are describing API-machinery related SLIs we would
+like to introduce and suggest which of those should eventually have a
+corresponding SLO replacing current "99% of API calls return in under 1s" one.
+
+The SLOs we are proposing in this doc are our goal - they may not be currently
+satisfied. As a result, while in the future we would like to block the release
+when we are violating SLOs, we first need to understand where exactly we are
+now, define and implement proper tests and potentially improve the system.
+Only once this is done, we may try to introduce a policy of blocking the
+release on SLO violation. But this is out of scope of this doc.
+
+
+### SLIs and SLOs proposal
+
+Below we introduce all SLIs and SLOs we would like to have in the api-machinery
+area. A bunch of those are not easy to understand for users, as they are
+designed for developers or performance tracking of higher level
+user-understandable SLOs. The user-oriented one (which we want to publicly
+announce) are additionally highlighted with bold.
+
+### Prerequisite
+
+Kubernetes cluster is available and serving.
```

Comment:
> The first questions is: do we need to define the target cluster size first? For example, below SLO applies to 2k nodes?

It's not only cluster size. There are a bunch of things that needs to be bounded. This is the doc that is initial attempt to address it:
https://github.com/kubernetes/community/blob/master/sig-scalability/thresholds.md

We are missing a bit more background for this doc, which is here: https://docs.google.com/document/d/15rD6XBtKyvXXifkRAsAVFBqEGApQxDRWM3H1bZSBsKQ/edit?pli=1#heading=h.ahomqwpyjhso
I will try to move parts of that doc to .md to provide more context some time in the future.

> And also, I am afraid using cloud provider VS baremetal also matters here. We may also need to clarify the cluster details first?

Cloud-provider VS baremetal doesn't matter for this particular thing at all. Though obviously how the cluster is set up matters (we won't satisfy SLO if e.g. master components will not have enough CPU). 
However, those aspects are being addressed in this doc:
https://github.com/kubernetes/community/blob/master/sig-scalability/provider-configs.md

There obviously might be missing informations there, but setup definitions is not in scope for this doc.


So to summarize - you have valid questions, but both are supposed to be addressed in separate docs (their initial versions already exist). We definitely can't put everything into a single doc, because then this doc won't be managable (I think we can easily get to 100 pages relatively soon).